package com.gridless.mixin;

import net.minecraft.client.gui.components.events.GuiEventListener;
import net.minecraft.client.gui.narration.NarratableEntry;
import net.minecraft.client.gui.screens.Screen;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.gen.Invoker;

@Mixin(Screen.class)
public interface ScreenAccessor {
    @Invoker("addRenderableWidget")
    <T extends GuiEventListener & net.minecraft.client.gui.components.Renderable & NarratableEntry> T invokeAddRenderableWidget(T widget);

    @Invoker("removeWidget")
    void invokeRemoveWidget(GuiEventListener widget);
}
